lua/spectre_common.lua

```diff
@@ -21,6 +21,7 @@ local HEADERS = {
     CACHE_STATUS = 'Spectre-Cache-Status',
     B3_TRACEID = 'X-B3-TraceId',
     ZIPKIN_ID = 'X-Zipkin-Id',
+    DOWNSTREAM_ERROR = 'X-Downstream-Error'
```

Comment:
nit, naming wise, this feels more like `X-Original-Status` or something. Would it make sense to just always attach such a header? I can imagine we'd want to look for cases where X-Original-Status and the status code the client received differ. However, since the failure mode needs to include connect timeouts to spectre, etc., we need to do this outside of spectre itself. Maybe in the spectre clientlib we can look at the delta per request, instead of trying to join the 2 requests back together in splunk later on?

Or did you have a different way to use this coming to mind?

Reply:
Yeah it does seem reasonable to just always include the header, and the name was very much a placeholder, essentially originating from a discussion between @sarguru and myself. X-Original-Status is a nicer name.